crm_location/tests/test_crm_location.py

```diff
@@ -38,9 +43,9 @@ def test_on_change_city(self):
         lead.location_id = location.id
         lead.on_change_city()
         self.assertEqual(lead.zip, location.name)
-        self.assertEqual(lead.city, location.city)
-        self.assertEqual(lead.state_id, location.state_id)
-        self.assertEqual(lead.country_id, location.country_id)
+        self.assertEqual(lead.city, location.city_id.name)
```

Comment:
This test is not well designed, as we are repeating the implementation in the test itself. We should assert data, not "code". Not sure if you follow me...

What we have to put here is:
```python
self.assertEqual(lead.city, 'Ghotam`)
```
Why? Because on current way, we are making the test dependent of the implementation, and the proof is that you needed to change it when the code has changed.

Please replace all of these occurrences.